packages/flutter/lib/src/painting/colors.dart

```diff
@@ -196,6 +212,211 @@ class HSVColor {
   String toString() => 'HSVColor($alpha, $hue, $saturation, $value)';
 }
 
+/// A color represented using [alpha], [hue], [saturation], and [lightness].
+///
+/// An [HSLColor] is represented in a parameter space that's motivated by human
+/// perception. The representation is useful for some color computations (e.g.,
+/// rotating the hue through the colors of the rainbow).
+///
+/// See also:
+///
+///   * [HSVColor], a color where the brightness is represented by value
+///     instead of lightness.
+@immutable
+class HSLColor {
+  /// Creates a color.
+  ///
+  /// All the arguments must not be null and be in their respective ranges. See
+  /// the fields for each parameter for a description of their ranges.
+  const HSLColor.fromAHSL(this.alpha, this.hue, this.saturation, this.lightness)
+    : assert(alpha != null),
+      assert(hue != null),
+      assert(saturation != null),
+      assert(lightness != null),
+      assert(alpha >= 0.0),
+      assert(alpha <= 1.0),
+      assert(hue >= 0.0),
+      assert(hue <= 360.0),
+      assert(saturation >= 0.0),
+      assert(saturation <= 1.0),
+      assert(lightness >= 0.0),
+      assert(lightness <= 1.0);
+
+  /// Creates an [HSLColor] from an RGB [Color].
+  ///
+  /// This constructor does not necessarily round-trip with [toColor] because
+  /// of floating point imprecision.
+  factory HSLColor.fromColor(Color color) {
+    final double alpha = color.alpha / 0xFF;
+    final double red = color.red / 0xFF;
+    final double green = color.green / 0xFF;
+    final double blue = color.blue / 0xFF;
+
+    final double max = math.max(red, math.max(green, blue));
+    final double min = math.min(red, math.min(green, blue));
+    final double delta = max - min;
+    final double lightness = (max + min) / 2.0;
+
+    double hue = 0.0;
+
+    if (max == 0.0) {
+      hue = 0.0;
+    } else if (max == red) {
+      hue = 60.0 * (((green - blue) / delta) % 6);
+    } else if (max == green) {
+      hue = 60.0 * (((blue - red) / delta) + 2);
+    } else if (max == blue) {
+      hue = 60.0 * (((red - green) / delta) + 4);
+    }
+
+    /// fix hue to 0.0 when red == green == blue.
+    hue = hue.isNaN ? 0.0 : hue;
+    final double saturation = max == 0.0 ? 0.0 : delta / max;
```

Comment:
according to the wikipedia article, Saturation is different for HSL and HSV, but you compute it the same in both cases.